ably/realtime/connection.py

```diff
@@ -213,7 +256,10 @@ async def close(self):
             log.warning('ConnectionManager: called close with no connected transport')
         self.enact_state_change(ConnectionState.CLOSED)
         if self.transport and self.transport.ws_connect_task is not None:
-            await self.transport.ws_connect_task
```

Comment:
I think you should add this to the close method to make sure you're not leaving any dangling tasks:
```python
if self.__ttl_task and not self.__ttl_task.done():
    self.__ttl_task.cancel()
```